opm/simulators/linalg/PressureBhpTransferPolicy.hpp

```diff
@@ -0,0 +1,270 @@
+/*
+  Copyright 2019 SINTEF Digital, Mathematics and Cybernetics.
+  Copyright 2019 Dr. Blatt - HPC-Simulation-Software & Services.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <opm/simulators/linalg/twolevelmethodcpr.hh>
+#include <dune/istl/paamg/pinfo.hh>
+
+namespace Opm
+{
+    template <class Communication>
+    void extendCommunicatorWithWells(const Communication& comm,
+                                     std::shared_ptr<Communication>& commRW,
+                                     const int nw)
+    {
```

Comment:
Have you looked at this function @blattms? If not, I would appreciate if you can check that it looks ok.